Neo4j.Driver/Neo4j.Driver/Internal/Helpers/TemporalHelpers.cs

```diff
@@ -16,92 +16,260 @@
 // limitations under the License.
 
 using System;
+using HdrHistogram;
 using HdrHistogram.Utilities;
+using Neo4j.Driver.Internal.Protocol;
+using Neo4j.Driver.Internal.Types;
 using Neo4j.Driver.V1;
 using TimeZoneConverter;
 
 namespace Neo4j.Driver.Internal
 {
     internal static class TemporalHelpers
     {
-        public const int NanosecondsPerTick = 100;
+        public const int MinYear = -999_999_999;
```

Comment:
@lutovich Can you also have a look at the new crazy methods added in this class?
Ref: https://github.com/ThreeTen/threetenbp